lib/segment/src/index/struct_payload_index.rs

```diff
@@ -175,19 +177,23 @@ impl StructPayloadIndex {
         payload_schema: PayloadFieldSchema,
     ) -> OperationResult<Vec<FieldIndex>> {
         let payload_storage = self.payload.borrow();
-        let mut field_indexes = index_selector(field, &payload_schema, self.db.clone(), true);
-        for index in &field_indexes {
-            index.recreate()?;
+        let mut builders = index_builder_selector(field, &payload_schema, self.db.clone());
+        for index in &mut builders {
+            index.init()?;
```

Comment:
We can remove `recreate` from all field indices now

Reply:
resolve this thread because we need this changes asap and `recreate` exists in tests